src/solvers/dgsem_p4est/dg_2d.jl

```diff
@@ -364,20 +363,15 @@ end
 
     # Call pointwise numerical flux function for the conservative part
     # in the normal direction on the boundary
-    flux_ = boundary_condition(u_inner, normal_direction, x, t, surface_flux, equations)
-
-    # Compute pointwise nonconservative numerical flux at the boundary.
-    noncons_ = boundary_condition(u_inner, normal_direction, x, t, nonconservative_flux,
-                                  equations)
+    flux_ = boundary_condition(u_inner, normal_direction, x, t,
+                               surface_integral.surface_flux, equations)
 
     # Copy flux to element storage in the correct orientation
     for v in eachvariable(equations)
         # Note the factor 0.5 necessary for the nonconservative fluxes based on
         # the interpretation of global SBP operators coupled discontinuously via
         # central fluxes/SATs
```

Comment:
I am also not very content that we have to expose this to the user in order to specify the boundary condition, but I don't see a better way to accomplish this. I definitely think that we should comment this to give some explanation where the factor `0.5` comes from and that it is not specific to any boundary condition.

Reply:
I agree with all of you. I'm also not a huge fan of the factor `0.5`. We may alternatively return a tuple from the boundary conditions for non-conservative systems and multiply by 0.5 inside the solver. As an example:
`   # Call pointwise numerical flux function for the conservative part
    # in the normal direction on the boundary
    flux_, noncons_ = boundary_condition(u_inner, normal_direction, x, t, surface_integral.surface_flux, equations)

    # Copy flux to element storage in the correct orientation
    for v in eachvariable(equations)
        # Note the factor 0.5 necessary for the nonconservative fluxes based on
        # the interpretation of global SBP operators coupled discontinuously via
        # central fluxes/SATs
        surface_flux_values[v, node_index, direction_index, element_index] = flux_[v] + 0.5f0 * noncons_[v]
    end`

Reply:
I like this idea, since it doesn't require knowledge about the implementation aspect of adding the `0.5` to create the boundary condition. It should work well for the BCs that are currently implemented, but I am not sure what this would look like for the new type of boundary condition that you plan to add. Would you then only set the `flux_` part and set the `noncons_` part to zero?
I think it would be helpful to include a specific example with such a new BC in the PR to better understand and evaluate what the implementation should look like.

Reply:
I also like this idea with the documentation of how it would work in practice from Patrick's comment above. That is, for something like the standard shallow water equations the jump in the bottom topography is zero at the physical boundary (typically) so one's new boundary condition could compute the conservative flux pieces and then have
```julia
   return flux_, SVector(0,0,0,0) # flux, noncons
```